impl/src/main/java/org/jboss/weld/contexts/unbound/RequestContextImpl.java

```diff
@@ -50,6 +51,9 @@ public void deactivate() {
     @Override
     public void clearAndSet(Collection<ContextualInstance<?>> setOfInstances) {
         getBeanStore().clear();
+        // invalidate caches for req., session, conv. scopes
+        // this might be needed for propagation on the thread where there are existing contexts
+        RequestScopedCache.invalidate();
```

Comment:
To be fair, I am not sure this is really needed and it sure isn't tested within this PR because those Arq. tests will always use an `AbstractBoundContext` implementation (either `Bound` or pure `HTTP` backed based on protocol used).
Even in SE tests the `Bound` variants will be in place.

@mkouba any thoughts?

Reply:
Better safe than sorry ;-)

Reply:
Yea, that was my thinking